vmdb/spec/models/ems_refresh/refreshers/foreman_refresher_spec.rb

```diff
@@ -0,0 +1,96 @@
+require "spec_helper"
+
+describe EmsRefresh::Refreshers::ForemanRefresher do
+  let(:provider) do
+    _guid, _server, zone = EvmSpecHelper.create_guid_miq_server_zone
+    FactoryGirl.create(:provider_foreman,
+                       :zone       => zone,
+                       :url        => "10.8.96.102",
+                       :verify_ssl => false,
+                      )
+  end
+
+  it "will perform a full refresh on api v2" do
+    VCR.use_cassette("#{described_class.name.underscore}_api_v2") do
+      EmsRefresh.refresh(provider)
```

Comment:
In most cases we run the refresh twice before checking the data to verify that the second run doesn't change the numbers (creating duplicate records).  We'll probably want to do something similar, but maybe run configuration_manager, check some values, run provisioning_manager, check some values, run configuration_manager again and verify that relationships now exist and values didn't change.